src/ir/ty.rs

```diff
@@ -104,9 +104,13 @@ impl Type {
         Self::new(Some(name), None, kind, false)
     }
 
-    pub fn is_integer_literal(&self) -> bool {
+    pub fn is_integer(&self, ctx: &BindgenContext) -> bool {
         match *self.kind() {
             TypeKind::Int(..) => true,
+            TypeKind::ResolvedTypeRef(inner) |
+            TypeKind::Alias(_, inner) => {
```

Comment:
Instead of modifying this, can you use `t.canonical_type(ctx).is_integer_literal()` instead?

Also, could you add a test for this?